straxen/plugins/acqmon_processing.py

```diff
@@ -240,6 +242,160 @@ def compute(self, aqmon_hits, start, end):
         result = result[sort]
         return result
 
+@export
+class GPS_sync(strax.Plugin):
+    """
+    Correct the event times to GPS time. 
+      1. Finds the TTL GPS pulses coming into the AM from the gps 
+    module and their pairs in the ASCII file coming from the module
+    for the correspondant run.
+      2. Corrects the timestamp of all events by linearly interpolating
+    between the previous and next sync pulses. 
+    """
+        
+    __version__ ='0.1.0'
+    depends_on = ('aqmon_hits', 'event_info')
+    provides  = 'gps_sync'
+    data_kind = 'gps_sync'
+    
+    def infer_dtype(self):
+        dtype = [(('Time since unix epoch [ns]', 'time'), np.int64),
+                 (('GPS absolute time [ns]', 't_gps'), np.int64),
+                ]
+        return dtype
+
+
+    def gps_times_from_runid(self,run_id):
+        '''
+        Fetches the mongodb looking for the gps_sync collection for 
+        timestamps between the start and end times of a given run_id.
+        '''
+
+        rundb = utilix.xent_collection()
+        gps_times = utilix.xent_collection(collection='gps_sync',  database='xenonnt')
+
+        if isinstance(run_id, str):
+            run_id = int(run_id)
+
+        query = {"number": run_id}
+
+        doc = rundb.find_one(query, projection={'start': 1, 'end': 1})
```

Comment:
probably should assert that the doc is not None here (if there is no run that matches the run_id doc will be None)